### PR TITLE
Upgrade ginkgo to v2

### PR DESCRIPTION
**Pull request by jsliacan**
**Fixes:** Issue #3258 

Integration tests should run as before.

Output file seems to have changed structure, might need to specify some options in reporter config.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -203,7 +203,7 @@ ifndef PULL_SECRET_PATH
 export PULL_SECRET_PATH = $(HOME)/Downloads/crc-pull-secret
 endif
 ifndef BUNDLE_PATH
-export BUNDLE_PATH = $(HOME)/Downloads/crc_libvirt_$(OPENSHIFT_VERSION).$(BUNDLE_EXTENSION)
+export BUNDLE_PATH = $(HOME)/Downloads/crc_libvirt_$(OPENSHIFT_VERSION)_amd64.$(BUNDLE_EXTENSION)
```
Comment (praveenkumar) [inline Makefile:206]: @jsliacan This should be part of different commit since here you are adding the arch explicit.

Comment (jsliacan) [reply Makefile:206]: @praveenkumar Thanks for noticing. Should I remove it or create an `ARCH` var in expectation of arm or keep it hardcoded like it is?

Comment (praveenkumar) [reply Makefile:206]: @jsliacan better to use `GOARCH` env from the Makefile.

**Comment by praveenkumar** [comment]
@jsliacan integration tests failing on the CI with following error, which means it is not able to generate the Junit files.
```
 ------------------------------
[ReportAfterSuite] [FAILED] [0.001 seconds]
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo
  Failed to generate JUnit report:
  open out/integration.xml: no such file or directory
  In [ReportAfterSuite] at: /home/packer/crc/vendor/github.com/onsi/ginkgo/v2/reporting_dsl.go:127 
```

File: test/integration/testsuite_test.go [hunk 2]
```diff
@@ -31,6 +31,10 @@ func TestTest(t *testing.T) {
 
 	// fetch the current (reporter) config
 	_, reporterConfig := GinkgoConfiguration()
+	err := os.Mkdir("out", 0775)
```
Comment (praveenkumar) [inline test/integration/testsuite_test.go:34]: I should use `os.MkdirAll` because in case `out` directory already exist then this will through an error but `MkdirAll` will not.
```
// If path is already a directory, MkdirAll does nothing
// and returns nil.
```
